bentoml/_internal/bento/build_config.py

```diff
@@ -285,24 +192,84 @@ def write_to_bento(
 
 def _docker_options_structure_hook(d: t.Any, _: t.Type[DockerOptions]) -> DockerOptions:
```

Comment:
move this to a convertor, remove the `bentoml_cattr.register_structure_hook` below